Habitica Intents/TaskManager.swift

```diff
@@ -0,0 +1,179 @@
+import ReactiveSwift
+import Habitica_Models
+//import Result
+
+
+class TaskManager: BaseRepository<TaskLocalRepository>, TaskRepositoryProtocol {
+    static let shared = TaskManager()
+    
+    override init() {
+        super.init()
+        self.setupNetworkClient()
+        if let cid = AuthenticationManager.shared.currentUserId {
+            AuthenticationManager.shared.currentUserKey = AuthenticationManager.shared.localKeychain[cid]
+        }
+        print("Auth at start id, ", AuthenticationManager.shared.currentUserId)
+        print("Auth at start key, ", AuthenticationManager.shared.currentUserKey)
+    }
+    
+    @objc
+    func setupNetworkClient() {
+        /*
+         Comment for code review, TODO, remove:
+         This function is a direct copy of the function from AppDelegate, there
+         may be a better solution to handle the project inegration, to avoid the duplicate
+         code, but it's beyond what I know regarding swift project setup. Alternatively,
+         we could separate the code, but that doesn't seem like a call I should make.
+         - Chris Coffin
+         */
+        NetworkAuthenticationManager.shared.currentUserId = AuthenticationManager.shared.currentUserId
+        NetworkAuthenticationManager.shared.currentUserKey = AuthenticationManager.shared.currentUserKey
+        updateServer()
+        //AuthenticatedCall.errorHandler = HabiticaNetworkErrorHandler()
+        let configuration = URLSessionConfiguration.default
+        //NetworkLogger.enableLogging(for: configuration)
+        AuthenticatedCall.defaultConfiguration.urlConfiguration = configuration
+    }
+    
+    func updateServer() {
+        /*
+         Comment for code review, TODO, remove:
+         This function is a direct copy of the function from AppDelegate, there
+         may be a better solution to handle the project inegration, to avoid the duplicate
+         code, but it's beyond what I know regarding swift project setup. Alternatively,
+         we could separate the code, but that doesn't seem like a call I should make.
+         - Chris Coffin
+         */
+        if let chosenServer = UserDefaults().string(forKey: "chosenServer") {
+            switch chosenServer {
+            case "staging":
+                AuthenticatedCall.defaultConfiguration = HabiticaServerConfig.staging
+            case "beta":
+                AuthenticatedCall.defaultConfiguration = HabiticaServerConfig.beta
+            case "gamma":
+                AuthenticatedCall.defaultConfiguration = HabiticaServerConfig.gamma
+            case "delta":
+                AuthenticatedCall.defaultConfiguration = HabiticaServerConfig.delta
+            default:
+                AuthenticatedCall.defaultConfiguration = HabiticaServerConfig.production
+            }
+        }
+    }
+    
+    func getTasks(predicate: NSPredicate, sortKey: String = "order") -> SignalProducer<ReactiveResults<[TaskProtocol]>, ReactiveSwiftRealmError> {
```

Comment:
Have you tried making the TaskRepositoryImpl available for this target? You should be able to do that by selecting the `Habitica Intents` target when looking at the following view in the inspector: 
![Screenshot 2019-05-15 at 14 44 22](https://user-images.githubusercontent.com/298062/57776529-08b13200-7720-11e9-9802-91b153385fda.png)

Reply:
Thanks, I'd done the same for a few other files, but I recall having an issue early on with including the TaskRepositoryImpl for some reason. I'll try again now that everything else is in place.  You're right that it would mean less duplicate code.

Reply:
I tried this again, and realized why I backed away. Including TaskRepositoryImpl caused me to have to include a chain of quite a bit of other code, including code centered around the UI. I figured including the two functions here directly would be easier to maintain. Would it make sense to split out some of the less general code into another file (e.g references to ToastManager)?